docs/website/scripts/integration-index/generate.js

```diff
@@ -0,0 +1,249 @@
+            write(' from ')
+            write('<a href="' + org.link + '">')
+            write(org.name)
+            write('</a>')
+        } else {
+            write(' from ')
+            write(speakers[i].organization)
+        }
+    }
+}
+
+function writeBlog(x) {
+    writeLink(x, x)
+}
+
+function writeTutorial(tutorial) {
+    writeLink(tutorial, tutorial)
+}
+
+function writeCode(code) {
+    writeLink(code, code)
+}
+
+function writeIndex() {
+    var layers = new Set();
+    writeln("## Index of Enforcement Points by Layer of the Stack")
+    Object.keys(enforcements).forEach(function(key) { 
+        c = enforcements[key]
+        if ('labels' in c && 'layer' in c.labels) {
+            layers.add(c.labels.layer)
+        } else {
+            layers.add(undefined)
+        }
+    })
+    layers.forEach(writeIndexLayer)
+
+}
+
+function writeIndexLayer(layer) {
+    if (layer == undefined) {
+        writeln('\n**unspecified**')
+    } else {
+        writeln("\n**" + layer + "**")
+    }
+    // sort by title name
+    keys = Object.keys(config.integrations)
+    objs = []
+    for (var i = 0; i < keys.length; i++) {
+        objs.push({key: keys[i], obj: config.integrations[keys[i]]})
+    }
+    objs.sort((a,b) =>  a.obj.title.localeCompare(b.obj.title))  
+    objs.forEach(function (x) {
+        key = x.key
+        integ = x.obj
+        if (getLayer(integ) != layer) {
+            return
+        }
+        write("* ")
+        writeLink(integ.title, "#" + key)
+        writeln('')
+    })
+}
+
+function getLayer(integ) {
+    try {
+        return integ.labels.layer
+    } catch (e) {
+        return undefined
+    }
+}
+function findTools() {
+    d = {}
+    Object.keys(config.integrations).forEach(function(key) { 
+        c = config.integrations[key]
+        if ('labels' in c && 'type' in c.labels && c.labels.type == 'poweredbyopa') {
+            d[key] = c
+        }
+    })  
+    return d
+}
+
+function findEnforcements() {
+    d = {}
+    Object.keys(config.integrations).forEach(function(key) { 
+        c = config.integrations[key]
+        if ('labels' in c && 'layer' in c.labels && (!('type' in c.labels) || c.labels.type == 'enforcement')) {
+            d[key] = c
+        }
+    })  
+    return d
+}
+
+function writeTools() {
+    var layers = new Set();
+    writeln("## Index of Tools Powered by OPA")
+    keys = Object.keys(tools)
+    for (var i = 0; i < keys.length; i++) {
+        if (i > 0) {
+            write(", ")
+        }
+        writeTool(keys[i])
+    }
+}
+
+function writeTool(id) {
+    write("* ")
+    writeLink(tools[id].title, "#" + key)
+    writeln('')
+}
+
+function writeHeader() {
+    stream.write("# Integration Index (Total " + Object.keys(config.integrations).length + ")\n")
+}
+
+function writeln(string) {
+    stream.write(string + '\n')
+}
+function write(string) {
+    stream.write(string)
+}
+
+
+tools = findTools()
+enforcements = findEnforcements()
```

Comment:
Could we move these up with the other consts defined at the top of the file? Or maybe like merge them into the config when initializing things? I just get a little nervous with globals floating around like this, IMO it makes it harder to find/fix issues later on.

Reply:
I replaced the global variables with repeated calls to the function.